internal/digreflect/func.go

```diff
@@ -45,8 +45,21 @@ type Func struct {
 
 // String returns a string representation of the function.
 func (f *Func) String() string {
-	// "path/to/package".MyFunction (path/to/file.go:42)
-	return fmt.Sprintf("%q.%v (%v:%v)", f.Package, f.Name, f.File, f.Line)
+	return fmt.Sprint(f)
+}
+
+// Format implements fmt.Formatter for Func, printing a single-line
+// representation for %v and a multi-line one for %+v.
+func (f *Func) Format(w fmt.State, c rune) {
+	if w.Flag('+') && c == 'v' {
+		// "path/to/package".MyFunction
+		// 	path/to/file.go:42
+		fmt.Fprintf(w, "%q.%v", f.Package, f.Name)
```

Comment:
Not a quite a nit, but looking at the [https://golang.org/pkg/fmt/](docs for fmt), it seems that there isn't an expectation of "+v" being being multiline. However, this is desirable in the vast majority of use cases -- with the only far fetched exception that I can think of being someone manually parsing the dig logging output and unexpectedly encountering a newline.

Reply:
Yeah there are no expectations on the output. Starting with multierr's errors, we've been taking the position that if needed, `%+v` is the "richly formatted" version. So this is just carrying that over here.